search_shards.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2012-present Oliver Eilhard. All rights reserved.
+// Use of this source code is governed by a MIT-license.
+// See http://olivere.mit-license.org/license.txt for details.
+
+package elastic
+
+import (
+	"context"
+	"fmt"
+	"github.com/olivere/elastic/uritemplates"
+	"net/url"
+	"strings"
+)
+
+// SearchShardsService computes a score explanation for a query and
+// a specific document.
+// See https://www.elastic.co/guide/en/elasticsearch/reference/6.0/search-shards.html
+type SearchShardsService struct {
+	client     *Client
+	pretty     bool
+	index      []string
+	routing    string
+	local      *bool
+	preference string
+}
+
+// NewSearchShardsService creates a new SearchShardsService.
+func NewSearchShardsService(client *Client) *SearchShardsService {
+	return &SearchShardsService{
+		client: client,
+	}
+}
+
+// Index sets the names of the indices to restrict the results.
+func (s *SearchShardsService) Index(index ...string) *SearchShardsService {
+	if s.index == nil {
```

Comment:
I think the `nil` check is not necessary. It was a misconception I had when I learned Go years ago :-)